network-store-integration-test/src/test/java/com/powsybl/network/store/integration/NetworkStoreIT.java

```diff
@@ -2230,6 +2551,24 @@ public void shuntCompensatorTest() {
         }
     }
 
+    @Test
+    public void testShuntCompensator() {
+        try (NetworkStoreService service = new NetworkStoreService(getBaseUrl())) {
+            Network network = NetworkStorageTestCaseFactory.create(service.getNetworkFactory());
+            service.flush(network);
+        }
+
+        try (NetworkStoreService service = new NetworkStoreService(getBaseUrl())) {
+            Map<UUID, String> networkIds = service.getNetworkIds();
+            assertEquals(1, networkIds.size());
+            Network readNetwork = service.getNetwork(networkIds.keySet().stream().findFirst().get());
+            assertEquals(2, readNetwork.getShuntCompensatorCount());
+            readNetwork.getShuntCompensator("SHUNT1").remove();
+            assertEquals(1, readNetwork.getShuntCompensatorCount());
+            service.flush(readNetwork);
+        }
+    }
```

Comment:
The test part where it is checked that the flush correctly persisted the removal is missing here.